src/Core/Form/IdentifiableObject/FormHandler.php

```diff
@@ -0,0 +1,242 @@
+
+/**
+ * Class FormHandler
+ */
+final class FormHandler implements FormHandlerInterface
+{
+    /**
+     * @var string
+     */
+    private $formType;
+
+    /**
+     * @var FormDataProviderInterface
+     */
+    private $dataProvider;
+
+    /**
+     * @var FormDataHandlerInterface
+     */
+    private $dataHandler;
+
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @var HookDispatcherInterface
+     */
+    private $hookDispatcher;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var bool
+     */
+    private $isDemoModeEnabled;
+
+    /**
+     * @param string $formType
+     * @param FormDataProviderInterface $dataProvider
+     * @param FormDataHandlerInterface $dataHandler
+     * @param FormFactoryInterface $formFactory
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param TranslatorInterface $translator
+     * @param bool $isDemoModeEnabled
+     */
+    public function __construct(
+        $formType,
+        FormDataProviderInterface $dataProvider,
+        FormDataHandlerInterface $dataHandler,
+        FormFactoryInterface $formFactory,
+        HookDispatcherInterface $hookDispatcher,
+        TranslatorInterface $translator,
+        $isDemoModeEnabled
+    ) {
+        $this->formFactory = $formFactory;
+        $this->dataProvider = $dataProvider;
+        $this->dataHandler = $dataHandler;
+        $this->hookDispatcher = $hookDispatcher;
+        $this->translator = $translator;
+        $this->formType = $formType;
+        $this->isDemoModeEnabled = $isDemoModeEnabled;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getForm(array $options = [])
+    {
+        return $this->buildForm(
+            $this->dataProvider->getDefaultData(),
+            null,
+            $options
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormFor($id, array $options = [])
+    {
+        return $this->buildForm(
+            $this->dataProvider->getData($id),
+            $id,
+            $options
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(FormInterface $form)
+    {
+        return $this->handleForm($form);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleFor($id, FormInterface $form)
+    {
+        return $this->handleForm($form, $id);
+    }
+
+    /**
+     * @param mixed $data
+     * @param int|null $id
+     * @param array $options
+     *
+     * @return FormInterface
+     */
+    private function buildForm($data, $id = null, array $options = [])
+    {
+        $formBuilder = $this->formFactory->createBuilder($this->formType, $data, $options);
+
+        $this->hookDispatcher->dispatchWithParameters('action' . $formBuilder->getName() . 'FormBuilderModifier', [
+            'form_builder' => $formBuilder,
+            'data' => &$data,
+            'id' => $id,
```

Comment:
just to be consistent, since this method is used to create form for both create and update actions, i think its okay to pass it as a `null ` as well. it would allow developer to easily check whether it's a create or update action. if we didnt pass it, developer would have to check it by `isset($params['id']` method, so i prefer having same parameters all the time, wdyt?

Reply:
is it really worth it? the only difference would be just `id` param and developer would be forced to register 2 hooks all the time. :/